sql/core/src/test/scala/org/apache/spark/sql/jdbc/JDBCSuite.scala

```diff
@@ -209,6 +209,14 @@ class JDBCSuite extends SparkFunSuite
     conn.close()
   }
 
+  // Check whether the tables are fetched in the expected degree of parallelism
+  def checkNumPartitions(df: DataFrame, expectedNumPartitions: Int): Unit = {
+    val explain = ExplainCommand(df.queryExecution.logical, extended = true)
+    val plans = spark.sessionState.executePlan(explain).executedPlan
+    val expectedMsg = s"${JDBCOptions.JDBC_NUM_PARTITIONS}=$expectedNumPartitions"
```

Comment:
instead of matching string, can we call `collect` on the logical plan and get the `JDBCRelation` to check the partitions directly?